### PR TITLE
Fix GoPro extractor.

### PR DESCRIPTION
**Pull request by stilor**
**IMPORTANT**: PRs without the template will be CLOSED

The GoPro extractor fails even when used on the URLs recorded in the tests with an error like:

```
[GoPro] Extracting URL: https://gopro.com/v/KRm6Vgp2peg4e [GoPro] KRm6Vgp2peg4e: Downloading webpage
ERROR: [GoPro] KRm6Vgp2peg4e: KRm6Vgp2peg4e: Failed to parse JSON (caused by JSONDecodeError('Expecting \',\' delimiter in \'"shop_url"=>"https:/\': line 1 column 5120 (char 5119)')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/storage/avn/git/yt-dlp/yt_dlp/extractor/common.py", line 718, in extract
...
```

- The JSON in the webpage now contains semicolons as part of HTML entities such as `&quot;`. This makes the regexp used for matching truncate the JSON prematurely.
- The same entities are replaced with the respective characters by `clean_html`, resulting in an invalid JSON.
- One of the tests for the extractor fails because of unexpected `author` and `track` fields in the returned dictionary.

<details open><summary>Template</summary>

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))
</details>

### CODE REVIEW

File: yt_dlp/extractor/gopro.py [hunk 1]
```diff
@@ -58,7 +60,8 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, video_id)
 
         metadata = self._parse_json(
-            self._html_search_regex(r'window\.__reflectData\s*=\s*([^;]+)', webpage, 'metadata'), video_id)
+            self._search_regex(r'<script>window\.__reflectData\s*=\s*(.+)\s*;\s*</script>',
+                               webpage, 'metadata'), video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/gopro.py:64]:
```py
        metadata = self._search_json(r'<script>window\.__reflectData\s*=', ...)
```

should work better

File: yt_dlp/extractor/gopro.py [hunk 2]
```diff
@@ -23,6 +23,8 @@ class GoProIE(InfoExtractor):
             'upload_date': '20210919',
             'uploader_id': 'fireydive30018',
             'duration': 396062,
+            'track': '',
+            'artist': '',
```
Comment (pukkandan) [inline yt_dlp/extractor/gopro.py:27]: The track is not supposed to be an empty string. Extractor's returning wrong value. You don't have to fix the code in this PR, but don't add wrong value to test.
```suggestion
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/gopro.py (hunk 1 above)
Comment (stilor) [reply yt_dlp/extractor/gopro.py]: Done.

File: yt_dlp/extractor/gopro.py (hunk 2 above)
Comment (stilor) [reply yt_dlp/extractor/gopro.py]: Split the fix into a separate commit.

**Review by seproDev — CHANGES_REQUESTED** [review]
Also, please don't force push. The commits will be squashed upon merge.

File: yt_dlp/extractor/gopro.py [hunk 3]
```diff
@@ -57,8 +57,8 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
 
-        metadata = self._parse_json(
-            self._html_search_regex(r'window\.__reflectData\s*=\s*([^;]+)', webpage, 'metadata'), video_id)
+        metadata = self._search_json(
+            r'<script>window\.__reflectData\s*=', webpage, 'metadata', video_id)
```
Comment (bashonly) [inline yt_dlp/extractor/gopro.py:61]: is adding the `<script>` tag necessary?

I think we should at least do this
```suggestion
            r'<script>\s*window\.__reflectData\s*=', webpage, 'metadata', video_id)
```

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/gopro.py (hunk 3 above)
Comment (seproDev) [reply yt_dlp/extractor/gopro.py:61]:
```suggestion
            r'window\.__reflectData\s*=', webpage, 'metadata', video_id)
```

**Pull request merged**